### PR TITLE
[RNMobile] Correct colors usage in Button

### PR DESCRIPTION
**Pull request by lukewalczak**
## Description

Correct colors usage in `Button` after web changes.

ref to gb-mobile: https://github.com/wordpress-mobile/gutenberg-mobile/pull/2155
## How has this been tested?

Testing HTML:

```
<!-- wp:button {"customBackgroundColor":"#325766"} -->
<div class="wp-block-button"><a class="wp-block-button__link has-background" style="background-color:#325766">CUSTOM</a></div>
<!-- /wp:button -->

<!-- wp:button {"gradient":"luminous-dusk"} -->
<div class="wp-block-button"><a class="wp-block-button__link has-background has-luminous-dusk-gradient-background">GRADIENT</a></div>
<!-- /wp:button -->

<!-- wp:button {"backgroundColor":"secondary"} -->
<div class="wp-block-button"><a class="wp-block-button__link has-background has-secondary-background-color">PALETTE</a></div>
<!-- /wp:button -->

<!-- wp:button {"customTextColor":"#90318f"} -->
<div class="wp-block-button"><a class="wp-block-button__link has-text-color" style="color:#90318f">TEXT CUSTOM</a></div>
<!-- /wp:button -->

<!-- wp:button {"textColor":"background"} -->
<div class="wp-block-button"><a class="wp-block-button__link has-text-color has-background-color">TEXT PALETTE</a></div>
<!-- /wp:button -->
```

Paste that HTML on the mobile editor and compare it with that screenshot:
Screenshot 2020-04-15 at 11 05 15

Another way to test:

1. Open post on web
2. Add Buttons block and create a couple of Buttons inside:
	- Button with custom background color
	- Button with color from palette color
	- Button with gradient color
	- Button with text custom color 
	- Button with text color from palette color
3. Open HTML view
4. Remove `Buttons` container to have separated list of Button
5. Save draft
6. Open post on mobile app
7. Expect:
	- Button with custom background color - button should have the same background color
	- Button with color from palette - button should have gray fallback color
	- Button with gradient color - button should have a gradient color
	- Button with text custom color - button text should have text custom color
	- Button with text color from palette - button text should have white text

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

File: packages/block-library/src/button/edit.native.js [hunk 1]
```diff
@@ -382,7 +391,11 @@ class ButtonEdit extends Component {
 							this.onToggleButtonFocus( true )
 						}
 						__unstableMobileNoFocusOnMount={ ! isSelected }
-						selectionColor={ textColor.color || '#fff' }
+						onBlur={ () => {
```
Comment (pinarol) [inline packages/block-library/src/button/edit.native.js:394]: Why is the new "onBlur" necessary?

Comment (dratwas) [inline packages/block-library/src/button/edit.native.js:394]: Is this change necessary in that PR?

Comment (lukewalczak) [reply packages/block-library/src/button/edit.native.js:394]: Shouldn't be there 😕

Comment (lukewalczak) [reply packages/block-library/src/button/edit.native.js:394]: Shouldn't be there 😕

File: packages/block-library/src/button/edit.native.js [hunk 2]
```diff
@@ -152,14 +152,29 @@ class ButtonEdit extends Component {
...
-			// `backgroundColor` which should be set when we are able to resolve it
+		const { backgroundColor, attributes } = this.props;
+		const { style } = attributes;
+
+		if ( style && style.color && style.color.background ) {
+			return style.color.background;
+		} else if ( backgroundColor.color ) {
 			return backgroundColor.color;
 		}
 		return styles.fallbackButton.backgroundColor;
```
Comment (dratwas) [inline packages/block-library/src/button/edit.native.js:163]:
```suggestion
		return (
			( style && style.color && style.color.background ) ||
			backgroundColor.color ||
			styles.fallbackButton.backgroundColor
		);
```

File: packages/block-library/src/button/edit.native.js [hunk 3]
```diff
@@ -382,7 +391,7 @@ class ButtonEdit extends Component {
 							this.onToggleButtonFocus( true )
 						}
 						__unstableMobileNoFocusOnMount={ ! isSelected }
-						selectionColor={ textColor.color || '#fff' }
+						selectionColor={ this.getTextColor() }
```
Comment (dratwas) [inline packages/block-library/src/button/edit.native.js:394]: I believe that is should be `getBackgroundColor` :)

Comment (lukewalczak) [reply packages/block-library/src/button/edit.native.js:394]: Nope, selection color depends on textColor

File: packages/block-library/src/button/edit.native.js (hunk 2 above)
Comment (lukewalczak) [reply packages/block-library/src/button/edit.native.js:163]: Ok, will do the same for `textColor`

**Review by dratwas — APPROVED** [review]
